File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
    if(nums.length==0) return 0
    
    const set = new Set();
    let max = 1
    
    for(let num of nums){
        set.add(num)
    }
    
    for(let num of nums){
        let count = 1
        
        let left = num - 1
        let right = num + 1
        
        while(set.has(left)){
            count++
            set.delete(left)
            left--        
        }
        
        while(set.has(right)){
            count++
            set.delete(right)
            right++        
        }
        
        max = Math.max(max,count)
        
    }
    
    return max
    
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
    
    if(nums.length==0) return 0;
    
    Set<Integer> set = new HashSet<Integer>();
    int max = 1;
    
    for(int num : nums){
        set.add(num);
    }
    
    for(int num : nums){
        int count = 1;
        
        int left = num - 1;
        int right = num + 1;
        
        while(set.contains(left)){
            count++;
            set.remove(left);
            left--;        
        }
        
        while(set.contains(right)){
            count++;
            set.remove(right);
            right++;        
        }
        
        max = Math.max(max,count);
        
    }
    
    return max;
    
    };
}